### ISSUE TITLE
Switching from One to other coloscheme breaks the latter

### ISSUE THREAD

**Issue by fedemp** [rakr/vim-one#9]
First select _one_. Then, when I switch to other coloschemes, mostly all syntax coloring is broken.

**Comment by keremc** [comment]
After changing the color scheme, do this: `:setfiletype ft` (substitute `ft` with the output of `:echo &ft`).

**Comment by rakr** [comment]
Hi, thanks for opening this issue. I've investigated this and it seems that it is not a problem that is directly related to he colorscheme, for instance solarized has the exact same problem. This seem to be related to the way Vim applies colorschemes.
Nonetheless, I'll this issue open and will try to reduce this side effect.
Thanks for you patience in the meantime.

**Comment by rakr** [comment]
@keremc, I was answering when you commented on this issue. Thanks for the tip.
What Ive read so far seems to be related to the way Vim handle the `hi def link` command when the colorscheme is defined.

**Comment by keremc** [comment]
@rakr Yep. It is a result of that and group links surviving `highlight clear`. You can fix it by making sure you set every group listed in the help pages `highlight-groups` and `group-name` from scratch. Don't do, for example, `highlight Comment guifg=#5c6370` but do `highlight Comment gui=NONE guibg=NONE guifg=#5c6370 guisp=NONE`. Link all groups from scratch. Then reload the syntax script. Not worth the hassle IMHO.

**Comment by keremc** [comment]
Disregard my last comment, it doesn't apply in this case. I think you should close this issue. There isn't much you can do to fix it. You can try to reduce the amount of language-specific highlighting rules but I believe that's pretty much all you can do.

**Comment by rakr** [comment]
Taking @keremc recommendation into account, I'm working on a version that reduces the amount of language-specific highlighting rules.
I'll push that to another branch from time to time, stay tuned, I'll push it to master once I'm satisfied with the result.
So far this is encouraging, some of the glitches that you describe @fedem are gone.

**Comment by cbaumhardt** [comment]
An alternative is to use [xolox/vim-colorscheme-switcher](https://github.com/xolox/vim-colorscheme-switcher) to switch colorschemes which takes care about this annoyance. Install this plugin (and its dependency `xolox/vim-misc`) and changing colorschemes is no longer a problem.

**Comment by rumpelsepp** [comment]
Just a suggestion; maybe this would fix the problem: https://github.com/jacoborus/estilo
Saw this in other vim colorschemes; you define your colors in YAML and the tool generates the VimL code.

**Comment by rakr** [comment]
Hi @rumpelsepp, thanks for mentioning this, **estilo** is an amazing project, the idea is definitely very cool and IMO, it's the best solution of this type (creating a template and generating the colorscheme).
I played with it and as a pet project tried to implement my other colorscheme using this (not published on Github yet). I also pushed a PR to support Elixir few months ago.

The features that I am missing which would make me definitely switch to estilo are:

* the ability to switch between dark and light background with the colorscheme being able to adapt
* some parameters like support for italic...

I guess that if this kind of feature finds its way into **estilo** I would happily migrate one, but not yet.

**Comment by rumpelsepp** [comment]
[This guy](https://github.com/mhartington/oceanic-next) here has also a light theme since he switched to estilo. I don't know how he does it, maybe you could have a quick look, since I am not that much used to colorscheme code for vim...
